src/System.Windows.Forms/tests/TestUtilities/DialogForm.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Windows.Forms;
+using Windows.Win32;
+using Windows.Win32.Foundation;
+using static Interop;
+
+namespace System;
+
+/// <summary>
+///  Helper class to host dialogs for testing.
+///  This class is typically passed as a parameter to a call to <see cref="CommonDialog.ShowDialog(IWin32Window?)"/>.
+/// </summary>
+internal class DialogForm : Form
+{
+    private User32.WM _message;
+    private User32.WM _action;
+
+    /// <summary>
+    ///  Constructs a DialogForm that performs <paramref name="action"/> to
+    ///  the dialog it is hosting once the <paramref name="message"/> has been received.
+    /// </summary>
+    /// <param name="message">
+    ///  The message the form should be listening for.
+    /// </param>
+    /// <param name="action">
+    ///  The action to be done to the dialog once the form has received <paramref name="message"/>.
+    /// </param>
+    public DialogForm(User32.WM message, User32.WM action)
```

Comment:
Further to the @Tanya-Solyanik feedback, if you want to allow this form to handle other messages, then you should do something like
```suggestion
    public DialogForm(Func<Message, bool> predicate, User32.WM messageToSend)
```
...where `predicate` for the above test would look something like:
```cs
using DialogForm dialogOwnerForm = new(m => m.MsgInternal == User32.WM.ENTERIDLE && m.WParamInternal == (uint)MSGF.DIALOGBOX, User32.WM.CLOSE);
```

Reply:
For now, I will leave it to only handle `WM_ENTERIDLE` as I'm currently not seeing other things we might want to test aside from confirming the dialog has displayed successfully. In the future we can make this more generic if needed, but wow this is very cool! Definitely keeping this in mind for the future. 😄